viewing_party/party.py

```diff
@@ -1,23 +1,125 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    if title and genre and rating:
+        movie_dict = {"title": title,
+        "genre": genre,
+        "rating": rating
+        }
+        return movie_dict
+    else:
+        return None
+
+def add_to_watched(user_data, movie):
+    watched_list = []
+    watched_list.append(movie)
+    user_data["watched"] = watched_list
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    watchlist = []
+    watchlist.append(movie)
+    user_data["watchlist"] = watchlist
+    return user_data
+
+def watch_movie(user_data, title):
+    for movie in user_data["watchlist"]:
+        if title == movie["title"]:
+            user_data["watchlist"].remove(movie)
+            user_data["watched"].append(movie)
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+    total = 0.0
+    movies = 0
+    for movie in user_data["watched"]:
+        total += movie["rating"]
+        movies += 1
+    if movies == 0:
+        avg = 0.0
+    else:
+        avg = total / movies
+    return avg
+
+def get_most_watched_genre(user_data):
+    count = 0
+    max_count = 0
+    max_genre = ""
+    for movie in user_data["watched"]:
+        genre = movie["genre"]
+        for movie in user_data["watched"]:
+            if genre in movie["genre"]:

```

Comment:
`movie["genre"]` and `genre` are both strings.  The `in` operator will return true if they are equal because it is checking to see if one string is contained within the other, but there is a potential bug here.  What if we had genres with similar names (ie, "Historical" and "Historical Drama").  This test would return true if the movie genre was "Historical Drama" and the genre being compared here was "Historical".  In this case, using `==` will fix this bug.